modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionStyleTest.ts

```diff
@@ -0,0 +1,271 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/directionality/Plugin';
+
+type Dir = 'rtl' | 'ltr';
+
+describe('browser.tinymce.plugins.directionality.DirectionStyleTest', () => {
+  const applyDir = (editor: Editor, dir: Dir) =>
+    TinyUiActions.clickOnToolbar(editor, dir === 'ltr' ? 'button[title="Left to right"]' : 'button[title="Right to left"]');
+
+  const testDirectionStyle = (editor: Editor, content: string, dir: Dir, expected: string) => {
+    editor.setContent(content);
+    TinySelections.setCursor(editor, [ 0, 0 ], 0);
+    applyDir(editor, dir);
+    TinyAssertions.assertContent(editor, expected);
+  };
+
+  const baseSettings = {
+    plugins: 'directionality',
+    toolbar: 'ltr rtl',
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  context('Inline direction style', () => {
+    const hook = TinyHooks.bddSetup<Editor>(baseSettings, [ Plugin ]);
+
+    it('TINY-9314: Applying ltr with dir="rtl" and direction: rtl', () => testDirectionStyle(
+      hook.editor(),
+      '<p dir="rtl" style="direction: rtl;">Lorem ipsum</p>',
+      'ltr',
+      '<p style="direction: ltr;">Lorem ipsum</p>'
```

Comment:
In this case it could unset the style and favor the attribute.

Reply:
This was a side effect favouring to keep the `direction` property if it was present initially. However, I am going to change the implementation to remove the `direction` property if it is not necessary, as detailed in the below conversation. The `dir` attribute will be removed regardless as its parent is in `ltr` already, to keep things minimal.